Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/TemplatedExporter/TemplatedExporterBase.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_EDITOR
+using Microsoft.Build.Unity.ProjectGeneration.Templates;
+using System;
+
+namespace Microsoft.Build.Unity.ProjectGeneration.Exporters.TemplatedExporter
+{
+    /// <summary>
+    /// Base class for file based exporters.
+    /// </summary>
+    internal class TemplatedExporterBase : TemplatedExporterPart
+    {
+        private readonly FileTemplate templateFile;
+        private readonly string exportPath;
+
+        protected TemplatedExporterBase(FileTemplate templateFile, string exportPath)
+            : base(templateFile.Root, templateFile.Root.CreateReplacementSet())
+        {
```

Comment:
nit: do we want to check whether templateFile or templateFile.Root is null? Or do we just want the NullPointerException to throw here?

Reply:
This is not necessarily public facing code, so no, just throw nullreferenceexception.